tests/integration/test_core.py

```diff
@@ -17,49 +17,56 @@ def test_sql_execution_chunked(ws, disposition):
     assert total == 1999999000000
 
 
-def test_sql_execution(ws, env_or_skip):
-    results = []
+NYC_TAXI_LIMITED_TRIPS = """
+WITH zipcodes AS (
+  SELECT DISTINCT pickup_zip, dropoff_zip 
+  FROM samples.nyctaxi.trips 
+  WHERE pickup_zip = 10282 AND dropoff_zip <= 10005
+)
+
+SELECT 
+  trips.pickup_zip, 
+  trips.dropoff_zip, 
+  trips.tpep_pickup_datetime, 
+  trips.tpep_dropoff_datetime 
+FROM 
+    zipcodes 
+  JOIN 
+    samples.nyctaxi.trips AS trips 
+    ON zipcodes.pickup_zip = trips.pickup_zip AND zipcodes.dropoff_zip = trips.dropoff_zip
+ORDER BY trips.dropoff_zip, trips.tpep_pickup_datetime, trips.tpep_dropoff_datetime 
+"""
+
+
+def test_sql_execution(ws, env_or_skip) -> None:
+    results = set()
     see = StatementExecutionExt(ws, warehouse_id=env_or_skip("TEST_DEFAULT_WAREHOUSE_ID"))
-    for pickup_zip, dropoff_zip in see.fetch_all(
-        "SELECT pickup_zip, dropoff_zip FROM nyctaxi.trips LIMIT 10", catalog="samples"
-    ):
-        results.append((pickup_zip, dropoff_zip))
-    assert results == [
-        (10282, 10171),
-        (10110, 10110),
-        (10103, 10023),
-        (10022, 10017),
-        (10110, 10282),
-        (10009, 10065),
-        (10153, 10199),
-        (10112, 10069),
-        (10023, 10153),
-        (10012, 10003),
-    ]
-
-
-def test_sql_execution_partial(ws, env_or_skip):
-    results = []
+    for pickup_zip, dropoff_zip, *_ in see.fetch_all(NYC_TAXI_LIMITED_TRIPS, catalog="samples"):
+        results.add((pickup_zip, dropoff_zip))
```

Comment:
To check the result-set precisely, `results` needs to remain a list.

Because the query orders the rows, order-sensitive equality checking is fine in this case.

It's not needed here, but in general for order-insensitive comparison the pattern is:
```python
results = […]
expected = {…}
assert len(results) == len(expected)
assert set(results) == expected
```

Reply:
I should clarify: right now this is performing neither an order-sensitive nor an order-insensitive check of the results, but rather something in-between.

In this case I think the comparison is supposed to be order-sensitive (because the query is), so please use lists for both `results` and the check against the expected set of results.

Reply:
The example you provided does not work as `assert len(results) == len(expected)` will assert to `False` as there are duplicates in `results`

Reply:
I think the integration test is just there to test we get some output back